src/custom/hooks/useApproveCallback/useApproveCallbackMod.ts

```diff
@@ -43,14 +52,19 @@ export function useApproveCallback(
     // we might not have enough data to know whether or not we need to approve
     if (!currentAllowance) return ApprovalState.UNKNOWN
 
+    // is the optionally set check amount less than current set allowance?
+    const isOptionalAmountLessThanAllowance = !!amountToCheckAgainstAllowance?.lessThan(currentAllowance)
+    // is current set allowance less than actual approving amount?
+    const isApprovingAmountLessThanAllowance = currentAllowance.lessThan(amountToApprove)
```

Comment:
David i think this is still wrong if im not mistaken 😅

`currentAllowance = "MAX Allowance in the bar" - 1 vinho verde`
`amountToApprove = "MAX Allowance in the bar"`

Therefore, u will get a `true` there as soon as you show up the second day in the bat

Reply:
Is it though? 

_Have never approved_
Line 56: is my selected investment amount LESS THAN my current set allowance? 
--> FALSE
Line 58: is my current set allowance (0 // undefined) LESS THAN the amount approving (ALWAYS MAX_INT)?
--> FALSE

Reply:
then typo in `L61`--> should be 
```ts
if (!isOptionalAmountLessThanAllowance || isApprovingAmountLessThanAllowance) {
```

Reply:
and variable names are funky, will change